nilearn/plotting/html_connectome.py

```diff
@@ -27,49 +65,149 @@ def _prepare_line(edges, nodes):
     return path_edges, path_nodes
 
 
-def _get_connectome(adjacency_matrix, coords, threshold=None,
-                    marker_size=None, cmap=cm.cold_hot, symmetric_cmap=True):
-    connectome = {}
-    coords = np.asarray(coords, dtype='<f4')
+def _prepare_colors(marker_color, number_of_nodes):
+    """
+    Generate "color" and "colorscale" attributes based on `marker_color` mode
+
+    Parameters
+    ----------
+    marker_color : color or sequence of colors, optional
+        Color(s) of the nodes. Default='auto'.
+
+    number_of_nodes : int
+        Number of nodes in the view
+
+    Returns
+    -------
+    markers_colors: list
```

Comment:
Describe a little bit the argument (although this is not strictly necessary in a private function)

Reply:
IIUC the logic of `to_color_strings` is duplicated here to avoid building and
querying a matplotlib colormap twice when `marker_color == "auto"`, is that the
case? If so, I am not sure this optimization is worth the code duplication (the
call to `to_color_strings` takes < 1 ms even with hundreds of nodes and thus is
small compared to `view_markers` execution time which is in the order of dozens
of ms, and of actually displaying the plot which is much longer).

also, WDYT of replacing the gray color of set2 with a different one? it is a bit
hard to see as it is the same color as the transparent brain